paddleseg/models/pfpnnet.py

```diff
@@ -0,0 +1,213 @@
+                    scale_head.append(
+                        nn.Upsample(scale_factor=2,
+                                    mode='bilinear',
+                                    align_corners=align_corners))
+            self.scale_heads.append(nn.Sequential(*scale_head))
+        self.scale_heads = nn.LayerList(self.scale_heads)
+
+        if self.enable_auxiliary_loss:
+            if dropout_ratio is not None:
+                self.dsn = nn.Sequential(
+                    layers.ConvBNReLU(fpn_inplanes[2],
+                                      fpn_inplanes[2],
+                                      3,
+                                      padding=1), nn.Dropout2D(dropout_ratio),
+                    nn.Conv2D(fpn_inplanes[2], num_class, kernel_size=1))
+            else:
+                self.dsn = nn.Sequential(
+                    layers.ConvBNReLU(fpn_inplanes[2],
+                                      fpn_inplanes[2],
+                                      3,
+                                      padding=1),
+                    nn.Conv2D(fpn_inplanes[2], num_class, kernel_size=1))
+
+        if dropout_ratio is not None:
+            self.dropout = nn.Dropout2D(dropout_ratio)
+        else:
+            self.dropout = None
+
+        self.conv_last = nn.Sequential(
+            layers.ConvBNReLU(len(fpn_inplanes) * fpn_dim,
+                              fpn_dim,
+                              3,
+                              bias_attr=False),
+            nn.Conv2D(fpn_dim, num_class, kernel_size=1))
+        self.conv_seg = nn.Conv2D(channels, num_class, kernel_size=1)
+
+    def cls_seg(self, feat):
+        if self.dropout is not None:
+            feat = self.dropout(feat)
+        output = self.conv_seg(feat)
+        return output
+
+    def forward(self, conv_out):
+        last_out = self.lateral_convs[-1](conv_out[-1])
+        f = last_out
+        fpn_feature_list = [last_out]
+        out = []
+        for i in reversed(range(len(conv_out) - 1)):
+            conv_x = conv_out[i]
+            conv_x = self.lateral_convs[i](conv_x)
+            prev_shape = paddle.shape(conv_x)[2:]
+            f = conv_x + F.interpolate(
+                f, prev_shape, mode='bilinear', align_corners=True)
+            fpn_feature_list.append(self.fpn_out[i](f))
+
+        output_size = fpn_feature_list[-1].shape[2:]
+
+        x = self.scale_heads[0](fpn_feature_list[-1])
+        for index in range(len(self.scale_heads) - 2, 0, -1):
+            x = x + F.interpolate(self.scale_heads[index](
+                fpn_feature_list[index]),
+                                  size=output_size,
+                                  mode='bilinear',
+                                  align_corners=self.align_corners)
+        x = self.cls_seg(x)
+        if self.enable_auxiliary_loss:
+            dsn = self.dsn(conv_out[2])
+            out.append(x)
+            out.append(dsn)
+            return out
```

Comment:
it seems line 188 can be removed and  line 209-211 can be replaced with: 
return [x, dsn]

Reply:
fix